thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/component.js

```diff
@@ -56,6 +68,29 @@ export default Component.extend({
    */
   selectedMode: Object.keys(ROOTCAUSE_MODE_MAPPING)[1],
 
+  /**
+   * Selected option for heatmap size
+   */
+  sizeOptionSelected: computed('sizeMetricUrn', 'sizeOptions', {
```

Comment:
can you not use this style for computed? Especially if you're not doing anything with set(). Try this
```
  sizeOptionSelected: computed(
    'sizeMetricUrn', 'sizeOptions',
    function() {
      const { sizeMetricUrn, sizeOptions } =
        this.getProperties('sizeMetricUrn', 'sizeOptions');
      return sizeOptions.find(opt => opt.urn === sizeMetricUrn);
    }
  ),
```

Reply:
try

```
 sizeOptionSelected: computed(
    'sizeMetricUrn', 'sizeOptions',
    function() {
      const { sizeMetricUrn, sizeOptions } =
        this.getProperties('sizeMetricUrn', 'sizeOptions');
      return sizeOptions.find(opt => opt.urn === sizeMetricUrn);
    }
  ).readOnly(),
```

Reply:
https://m.alphasights.com/the-risks-you-take-when-you-use-ember-computed-oneway-f2c008a14404

Reply:
this works. thanks for the example!

Reply:
Ideally we should use `oneWay` as readOnly will throw an error if someone (like a component) tries to mutate the object. OneWay makes a copy of the computed object and pass it on.